src/services/implementation/ClientServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package services.implementation;
+
+import entities.Client;
+import enums.StatusEnum;
+import exceptions.services.EntityNotFoundException;
+import exceptions.services.IdNullPointerException;
+import exceptions.services.NullEntityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import services.interfaces.IClientInterface;
+import util.ClientServiceUtility;
+import util.UserUtil;
+
+public class ClientServiceImpl implements IClientInterface {
+  private final List<Client> clients;
+  private Long currentId;
+
+  public ClientServiceImpl() {
+    this.clients = new ArrayList<>();
+    this.currentId = 0L;
+  }
+
+  @Override
+  public Optional<Client> insert(Client client) {
```

Comment:
When you insert a client status should be set to active, created date and updated date should be set to current date

Reply:
solved